src/mito2/src/manifest/storage.rs

```diff
@@ -661,6 +683,69 @@ impl ManifestObjectStore {
     fn dec_total_manifest_size(&self, val: u64) {
         self.total_manifest_size.fetch_sub(val, Ordering::Relaxed);
     }
+
+    /// Fetch all staging manifest files and return them as (version, action_list) pairs.
+    pub async fn fetch_staging_manifests(&self) -> Result<Vec<(ManifestVersion, Vec<u8>)>> {
+        let manifest_entries = self
+            .get_paths(
+                |entry| {
+                    let file_name = entry.name();
+                    if is_delta_file(file_name) {
+                        let version = file_version(file_name);
+                        Some((version, entry))
+                    } else {
+                        None
+                    }
+                },
+                true,
+            )
+            .await?;
+
+        let mut sorted_entries = manifest_entries;
+        Self::sort_manifests(&mut sorted_entries);
+
+        self.fetch_manifests_from_entries(sorted_entries).await
+    }
+
+    /// Clear all staging manifest files.
+    pub async fn clear_staging_manifests(&mut self) -> Result<()> {
+        let staging_entries = self
+            .get_paths(
+                |entry| {
+                    let file_name = entry.name();
+                    if is_delta_file(file_name) || is_checkpoint_file(file_name) {
+                        Some(entry)
+                    } else {
+                        None
+                    }
+                },
+                true,
+            )
+            .await?;
+
+        if staging_entries.is_empty() {
+            debug!("No staging manifest files to clear");
+            return Ok(());
+        }
+
+        let delete_tasks = staging_entries.iter().map(|entry| async {
+            debug!("Deleting staging manifest file: {}", entry.path());
+            self.object_store
+                .delete(entry.path())
+                .await
+                .context(OpenDalSnafu)
+        });
+
+        try_join_all(delete_tasks).await?;
```

Comment:
Can we use the deleter api?
https://docs.rs/opendal/latest/opendal/struct.Operator.html#method.deleter